easybuild/easyblocks/e/elpa.py

```diff
@@ -0,0 +1,227 @@
+                    self.cfg.update('configopts', '--enable-sse-assembly')
+                else:
+                    self.cfg.update('configopts', '--disable-sse')
+                    self.cfg.update('configopts', '--disable-sse-assembly')
+            elif flag == 'avx512f':
+                if getattr(self, 'avx512f'):
+                    self.cfg.update('configopts', '--enable-avx512')
+                else:
+                    self.cfg.update('configopts', '--disable-avx512')
+            else:
+                if getattr(self, flag):
+                    self.cfg.update('configopts', '--enable-%s' % flag)
+                else:
+                    self.cfg.update('configopts', '--disable-%s' % flag)
+
+        # By default ELPA tries to use MPI and configure fails if it's not available
+        # so we turn it off if MPI is not available
+        # We also set the LIBS environmet variable to detect the correct linalg library
+        # depending on the MPI availability.
+        if self.toolchain.options.get('usempi', None):
+            self.cfg.update('configopts', 'LIBS="$LIBSCALAPACK"')
+        else:
+            self.cfg.update('configopts', '--with-mpi=no')
+            self.cfg.update('configopts', 'LIBS="$LIBLAPACK"')
+
+        # make all builds verbose
+        self.cfg.update('buildopts', 'V=1')
+
+        # keep track of common configopts and of configopts specified
+        # in easyconfig file, so we can include them in each iteration
+        # later
+        common_config_opts = self.cfg['configopts']
+        common_build_opts = self.cfg['buildopts']
+
+        self.cfg['configopts'] = []
+        self.cfg['buildopts'] = []
+
+        with_omp_opts = [False]
+        if self.cfg['with_openmp']:
+            with_omp_opts.append(True)
+
+        for with_omp in with_omp_opts:
+            if with_omp:
+                omp_configopt = '--enable-openmp'
+            else:
+                omp_configopt = '--disable-openmp'
+
+            # append additional configure and build options
+            self.cfg.update('configopts',
+                            [omp_configopt + ' ' + common_config_opts])
+            self.cfg.update('buildopts', [common_build_opts])
+
+        self.log.debug("List of configure options to iterate over: %s", self.cfg['configopts'])
+        self.log.debug("List of build options to iterate over: %s", self.cfg['buildopts'])
+
+        return super(EB_ELPA, self).run_all_steps(*args, **kwargs)
+
+    def sanity_check_step(self):
+        """Custom sanity check for ELPA."""
+
+        custom_paths = {
+            'dirs': ['lib/pkgconfig', 'bin'],
+        }
+
+        shlib_ext = get_shared_lib_ext()
+
+        extra_files = []
+
+        with_omp_opts = [False]
+        if self.cfg['with_openmp']:
+            with_omp_opts.append(True)
+
+        if self.toolchain.options.get('usempi', None):
+            mpi_suff = ''
+        else:
+            mpi_suff = '_onenode'
```

Comment:
@micaeljtoliveira Please clarify what the `onenode` means exactly (i.e. no MPI support) with a comment above, since this is confusing. ;)